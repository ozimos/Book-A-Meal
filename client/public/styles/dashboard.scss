header.extra.header{
    margin-top: 20px
}