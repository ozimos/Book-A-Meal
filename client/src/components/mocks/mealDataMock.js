export const rows = [
  { id: 'abc', price: 1200 }, { id: 'def', price: 2000 }
];

export const allMeals = {
  data: { rows }

};

export const meal = {
  data: { id: 'mealId',
    price: 200,
    description: 'nice',
    imageUrl: 'nice',
    title: 'nice'
  }
};
