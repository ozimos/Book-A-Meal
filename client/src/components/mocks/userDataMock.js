export const userData = {
  data: { id: '123', email: 'asdf' },
  message: 'abc',
  token: 'abc'
};
export const userInput = {
  data: { email: 'asdf' }
};
export const userIncludes = {
  email: 'asdf',
  firstName: 'first',
  lastName: 'last'
};
