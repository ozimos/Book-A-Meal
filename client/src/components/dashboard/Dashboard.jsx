import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import ReactModal from 'react-modal';
import {
  Accordion,
  AccordionItem,
  AccordionItemTitle,
  AccordionItemBody,
} from 'react-accessible-accordion';
import SearchInput, { createFilter } from 'react-search-input';
import Greeting from '../greeting/Greeting';
import { dashboardActions } from '../../redux/actions';
import ConnectedNav from '../nav/Nav';
import uniqueUsers from '../../utils/uniqueUsers';
import cashTotal, { subTotal } from '../../utils/cashTotal';
import orderDetails from '../../utils/orderDetails';
import '../../../public/styles/dashboard.scss';

ReactModal.setAppElement(document.getElementById('root'));

class Dashboard extends React.Component {

  constructor(props) {
    super(props);

    this.state = {
      searchTerm: '',
    };
    this.searchUpdated = this.searchUpdated.bind(this);
  }
  componentDidMount() {
    this.props.dispatch(dashboardActions.getOrdersByDate());
  }
  searchUpdated(term) {
    this.setState({ searchTerm: term });
  }

  render() {
    const KEYS_TO_FILTERS = ['id', 'userId', 'updatedAt'];
    const isTodayOrder = this.props.orders.length !== 0;
    let filteredOrders;
    if (isTodayOrder) {
      filteredOrders = this.props.orders
        .filter(createFilter(this.state.searchTerm, KEYS_TO_FILTERS));
    }
    const { isCaterer, firstName } = this.props.user;
    return (
      <div id="dashboard" className="canvas2">
        <div className="container3">

          <header className="header extra">
            <ConnectedNav />
          </header>
          <main>
            <Greeting isCaterer={isCaterer} firstName={firstName} />

            <div className="flexbox">
              <div className="card order_summary">
                <div className="flexbox">
                  <span className="order_span summary_value flow-text">{this.props.orders.length}</span>
                  <span className="order_span shrink"><img alt="cutlery" className="responsive-img" src="./images/food.png" /></span>
                </div>
                <h5 className="flow-text" >Orders</h5>
              </div >
              <div className="card order_summary">
                <div className="flexbox">
                  <span className="order_span summary_value flow-text">{uniqueUsers(this.props.orders)}</span>
                  <span className="order_span shrink"><img alt="head profile" className="responsive-img" src="./images/profile.png" /></span>
                </div>
                <h5 className="flow-text">Customer(s)</h5>
              </div >
              <div className="card order_summary">
                <div className="flexbox">
                  <span className="order_span summary_value flow-text">&#8358;{cashTotal(this.props.orders)}</span>
                  <span className="order_span shrink"><img alt="upward stock chart" className="responsive-img" src="./images/chart.png" /></span>
                </div>
                <h5 className="flow-text">Total Sales</h5>
              </div>
            </div>
            <div className="order_detail flexbox">
              <h5>
                Today&#39;s Orders
              </h5>
              <SearchInput className="search-input input-field" onChange={this.searchUpdated} />
            </div>
            <Accordion accordion={false}>
              <AccordionItem expanded>
                <AccordionItemTitle />
                <AccordionItemBody>
                  <div>
                    { isTodayOrder ?
                      <table className="striped responsive-table">
                        <thead>
                          <tr>
                            <th>S/N</th>
                            <th>Customer</th>
                            <th>Order</th>
                            <th>Meals</th>
                            <th>Quantities</th>
                            <th>Total(&#8358;)</th>
                          </tr>
                        </thead>
                        <tbody>
                          {filteredOrders && filteredOrders.map((order, index) => (
                            <tr key={order.id}>
                              <td>{index + 1}</td>
                              <td>{`${order.User.firstName} ${order.User.lastName}`}</td>
                              <td>{order.id}</td>
                              <td>{orderDetails.mealTitleList(order)}</td>
                              <td>{orderDetails.mealQuantityList(order)}</td>
                              <td>{subTotal(order.Meals)}</td>
                            </tr>
                        ))
                      }
                        </tbody>
                      </table> :
                      <div>No orders have been placed today</div>
                  }
                  </div>
                </AccordionItemBody>
              </AccordionItem>
            </Accordion>
          </main>
        </div>
      </div>
    );
  }
}

Dashboard.propTypes = {
  dispatch: PropTypes.func.isRequired,
  orders: PropTypes.arrayOf(PropTypes.object).isRequired,
  user: PropTypes.shape({
    isCaterer: PropTypes.bool,
    firstName: PropTypes.string
  }).isRequired,
};
const mapStateToProps = state => ({
  user: state.loginReducer.user.data,
  meals: state.mealsReducer.meals,
  orders: state.dashboardReducer.orders,
  authenticated: state.loginReducer.authenticated,
});
export { Dashboard };
export default connect(mapStateToProps)(Dashboard);
