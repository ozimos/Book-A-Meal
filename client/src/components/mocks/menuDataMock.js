import { allMeals } from './mealDataMock';

export default {
  data: { Meals: allMeals.data }
};