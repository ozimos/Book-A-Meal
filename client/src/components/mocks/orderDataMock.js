import { allMeals } from './mealDataMock';

const Meals = allMeals.data.map(meal => ({ ...meal, MealOrders: { quantity: 2 } }));
export const order = {
  data: { id: 'orderId',
    userId: 'userId',
    Meals,
    mealList: ['abc', 'def'],
    quantityList: [2, 2] }
};
export const allOrders = {
  data: [{ ...order.data, id: 'order1' }, { ...order.data, id: 'order2' }]
};