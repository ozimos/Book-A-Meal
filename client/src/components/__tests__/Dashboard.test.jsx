/* global React:false, shallow:false toJson:false */

import { Dashboard } from '../dashboard/Dashboard';
import { allOrders } from '../mocks/orderDataMock';

describe('Dashboard Component', () => {
  const user = { isCaterer: true, firstName: 'user' };

  it('should render without throwing an error', () => {
    expect(shallow(<Dashboard
      dispatch={jest.fn()}
      user={user}
      orders={allOrders.data}
    />)
      .exists(<table className="table" />)).toBe(true);
  });
  it('renders correctly', () => {
    const wrapper = shallow(<Dashboard
      dispatch={jest.fn()}
      user={user}
      orders={allOrders.data}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

