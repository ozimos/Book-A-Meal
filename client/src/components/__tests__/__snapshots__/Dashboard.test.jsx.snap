// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Component renders correctly 1`] = `
<div
  className="canvas2"
  id="dashboard"
>
  <div
    className="container3"
  >
    <header
      className="header extra"
    >
      <Connect(Nav) />
    </header>
    <main>
      <Greeting
        firstName="user"
        isCaterer={true}
      />
      <div
        className="flexbox"
      >
        <div
          className="card order_summary"
        >
          <div
            className="flexbox"
          >
            <span
              className="order_span summary_value flow-text"
            />
            <span
              className="order_span shrink"
            >
              <img
                alt="cutlery"
                className="responsive-img"
                src="./images/food.png"
              />
            </span>
          </div>
          <h5
            className="flow-text"
          >
            Orders
          </h5>
        </div>
        <div
          className="card order_summary"
        >
          <div
            className="flexbox"
          >
            <span
              className="order_span summary_value flow-text"
            />
            <span
              className="order_span shrink"
            >
              <img
                alt="head profile"
                className="responsive-img"
                src="./images/profile.png"
              />
            </span>
          </div>
          <h5
            className="flow-text"
          >
            Customer(s)
          </h5>
        </div>
        <div
          className="card order_summary"
        >
          <div
            className="flexbox"
          >
            <span
              className="order_span summary_value flow-text"
            >
              â‚¦
            </span>
            <span
              className="order_span shrink"
            >
              <img
                alt="upward stock chart"
                className="responsive-img"
                src="./images/chart.png"
              />
            </span>
          </div>
          <h5
            className="flow-text"
          >
            Total Sales
          </h5>
        </div>
      </div>
      <div
        className="order_detail flexbox title-element"
      >
        <h5>
          Order History
        </h5>
        <Search
          caseSensitive={false}
          className="search-input input-field"
          fuzzy={false}
          onChange={[Function]}
          throttle={200}
        />
      </div>
      <AccordionWrapper
        accordion={false}
        className="accordion"
        onChange={[Function]}
      >
        <AccordionItemWrapper
          accordionStore={
            AccordionContainer {
              "_listeners": Array [],
              "state": Object {
                "accordion": true,
                "items": Array [],
                "onChange": [Function],
              },
            }
          }
          className="accordion__item"
          disabled={false}
          expanded={true}
          hideBodyClassName=""
        >
          <AccordionItemTitleWrapper
            className="accordion__title"
            hideBodyClassName=""
          />
          <AccordionItemBodyWrapper
            className="accordion__body"
            hideBodyClassName="accordion__body--hidden"
          >
            <div />
            <DashboardContainer
              getOrderMeals={[Function]}
              getOrderMealsTotals={[Function]}
              loading={false}
              onFetchData={[Function]}
              orders={
                Array [
                  Object {
                    "Meals": Array [
                      Object {
                        "MealOrders": Object {
                          "quantity": 2,
                        },
                        "id": "abc",
                        "price": 1200,
                      },
                      Object {
                        "MealOrders": Object {
                          "quantity": 2,
                        },
                        "id": "def",
                        "price": 2000,
                      },
                    ],
                    "User": Object {
                      "email": "asdf",
                      "firstName": "first",
                      "lastName": "last",
                    },
                    "id": "order1",
                    "userId": "userId",
                  },
                  Object {
                    "Meals": Array [
                      Object {
                        "MealOrders": Object {
                          "quantity": 2,
                        },
                        "id": "abc",
                        "price": 1200,
                      },
                      Object {
                        "MealOrders": Object {
                          "quantity": 2,
                        },
                        "id": "def",
                        "price": 2000,
                      },
                    ],
                    "User": Object {
                      "email": "asdf",
                      "firstName": "first",
                      "lastName": "last",
                    },
                    "id": "order2",
                    "userId": "userId",
                  },
                ]
              }
              pagination={
                Object {
                  "limit": 10,
                  "offset": 0,
                  "pages": 1,
                }
              }
            />
          </AccordionItemBodyWrapper>
        </AccordionItemWrapper>
      </AccordionWrapper>
    </main>
    <Modal
      ariaHideApp={true}
      bodyOpenClassName="ReactModal__Body--open"
      className="modal-content"
      closeTimeoutMS={0}
      contentLabel="Input Modal"
      isOpen={false}
      onRequestClose={[Function]}
      parentSelector={[Function]}
      portalClassName="ReactModalPortal"
      shouldCloseOnEsc={true}
      shouldCloseOnOverlayClick={true}
      shouldFocusAfterRender={true}
      shouldReturnFocusAfterClose={true}
      style={
        Object {
          "content": Object {
            "width": "60%",
          },
        }
      }
    >
      <MealsTable
        closeMealDetailModal={[Function]}
        getTrProps={[Function]}
        loading={false}
        meals={Array []}
        mealsPagination={
          Object {
            "limit": 5,
            "offset": 0,
            "pages": 1,
          }
        }
        onFetchData={[Function]}
        total={Object {}}
      />
    </Modal>
  </div>
</div>
`;
